python/cudf/cudf/core/multiindex.py

```diff
@@ -909,6 +909,112 @@ def from_product(cls, arrays, names=None):
         result = cls.from_pandas(pdi)
         return result
 
+    def _poplevels(self, level):
+        """
+        Pop the specified levels from self
+        """
+        if not pd.api.types.is_list_like(level):
+            level = (level,)
+
+        ilevels = [self._level_index_from_level(lev) for lev in level]
+
+        popped_data = OrderedDict({})
+        popped_names = []
+        names = list(self.names)
+
+        for i in sorted(ilevels, reverse=True):
+            names.pop(i)
+            n = self._data.names[i]
+            popped_data[n] = self._data.pop(n)
+            popped_names.insert(0, self.names[i])
+            popped_data.move_to_end(n, last=False)
```

Comment:
This still looks super fishy to me...

Reply:
What are your thoughts on https://github.com/rapidsai/cudf/pull/5779#discussion_r479334013?

Reply:
Closing this in favor of the other comment.